geode-assembly/src/integrationTest/java/org/apache/geode/management/internal/AgentUtilJUnitTest.java

```diff
@@ -50,4 +53,19 @@ public void testPulseWarExists() {
     String gemFireWarLocation = agentUtil.findWarLocation("geode-pulse");
     assertNotNull("Pulse WAR File was not found", gemFireWarLocation);
   }
+
+  @Test
+  public void testLookupOfWarFileOnClassPath() {
+    String classpath = System.getProperty("java.class.path");
+    String gemFireWarLocation = agentUtil.findWarLocation("testWarFile");
+    assertNull(gemFireWarLocation);
+
+    classpath =
+        classpath + System.getProperty("path.separator") + "somelocation/testWarFile.war";
+    System.setProperty("java.class.path", classpath);
```

Comment:
you probably would want to use a SystemPropertiesRestoreRule in this test to restore the system property afterwards since you changed it in this test.

Reply:
Already there :)